x509/cert_validations.go

```diff
@@ -25,8 +25,17 @@ func ValidateTimeStampingCertChain(certChain []*x509.Certificate, signingTime ti
 }
 
 func validateCertChain(certChain []*x509.Certificate, expectedLeafEku x509.ExtKeyUsage, signingTime time.Time) error {
-	if len(certChain) < 2 {
-		return errors.New("certificate chain must contain at least two certificates: a root and a leaf certificate")
+	if len(certChain) < 1 {
+		return errors.New("certificate chain must contain at least one certificates")
+	}
+
+	// For self-signed signing certificate (not a CA)
+	if len(certChain) == 1 {
+		cert := certChain[0]
+		if err := cert.CheckSignature(cert.SignatureAlgorithm, cert.RawTBSCertificate, cert.Signature); err != nil {
```

Comment:
Can we use `isSelfSigned` method below?

Reply:
We cannot use `isSelfSigned` as it internally calls `subject.CheckSignatureFrom(issuer)` that performs BasicConstrant and KeyUsageCertSign validations which are not valid for self signed leaf/signing certificate.